packages/@okta/vuepress-site/code/java/spring_security_saml/index.md

```diff
@@ -7,217 +7,372 @@ excerpt: >-
+### Run the app and authenticate
 
-> **Note:** Links in the app aren't functional yet because we haven't configured any IDPs. Full app functionality is completed after the **Configure Spring Security SAML to work with Okta** section.
+1. Run your Spring Boot app. You can do this from your IDE, or as follows using the command line:
 
-Here's what it should look like: ![App Running](/img/spring-security-saml-intro.png "App Running")
+   ```shell
+   ./gradlew bootRun
+   ```
 
-## Configure Okta to work with Spring Security SAML
+1. Open `http://localhost:8080` in your favorite browser and sign in with the credentials you used to create your account. You should see a successful result in your browser.
+
+   <div class="three-quarter border">
+
+   ![Login success](/img/saml/spring-security-saml-login.png)
+
+   </div>
+
+If you try to sign out, it won't work. You'll fix that in the next section.
+
+### Add a sign out feature
+
+Spring Security's SAML support has a [sign out feature](https://docs.spring.io/spring-security/reference/servlet/saml2/logout.html) that requires configuration.
+
+1. Edit your application on Okta and navigate to **General** > **SAML Settings** > **Edit**.
+
+1. Continue to the **Configure SAML** step and **Show Advanced Settings**. Before you can enable single logout, you'll have to create and upload a certificate to sign the outgoing logout request.
+
+1. You can create a private key and certificate using OpenSSL. Answer at least one of the questions with a value, and it should work.
+
+   ```shell
+   openssl req -newkey rsa:2048 -nodes -keyout local.key -x509 -days 365 -out local.crt
+   ```
+
+1. Copy the generated files to your app's `src/main/resources` directory. Configure `signing` and `singlelogout` in `application.yml`:
+
+   ```yaml
+   spring:
+     security:
+       saml2:
+         relyingparty:
+           registration:
+             okta:
+               signing:
+                 credentials:
+                   - private-key-location: classpath:local.key
+                     certificate-location: classpath:local.crt
+               singlelogout:
+                 binding: POST
+                 response-url: "{baseUrl}/logout/saml2/slo"
+               assertingparty:
+                 metadata-uri: <your-metadata-uri>
+   ```
+
+1. Upload the `local.crt` to your Okta app. Select *Enable Single Logout* and use the following values:
+
+   * Single Logout URL: `http://localhost:8080/logout/saml2/slo`
+   * SP Issuer: `http://localhost:8080/saml2/service-provider-metadata/okta`
+
+1. Finish configuring your Okta app, restart your Spring Boot app, and the logout button should work.
+
+   <div class="three-quarter border">
+
+   ![Sign out success](/img/saml/spring-security-saml-logout-success.png)
```

Comment:
It doesn't make sense for this diagram to have such different styling to all the others. Can you retake it please without all the styling applied to amtch the others?

Reply:
Or applying the same styling to the others to match this one?

Reply:
I didn't add any styling to this image. You can see the screenshot I took in the blog post. I just copied it into this repo. https://developer.okta.com/blog/2022/08/05/spring-boot-saml

Reply:
Ok that's just a weirdness in the generated app then if they don't have styles for the og in page but do for the log out screen.

Reply:
I'm just going to leave the screenshots as-is, in the interest of getting these updates over the finish line.